### PR TITLE
Add support for TableFunctionManager parameters in UDTFs.

### PR DESCRIPTION
**Pull request by brenocfg**
Solves #372.

As the title says, this adds support for OmnisciDB UDTFs with `TableFunctionManager` parameters, which allow thread-safe memory/exception management.

This PR implements an `OmnisciTableFunctionManager` custom type, which supports calls to `error_message(const char*)` and `set_output_row_size(int64)` methods. Similarly to #367, we require inputs to `error_message()` to be string literals.

Approving/merging this depends on [OmnisciDB PR 6035](https://github.com/omnisci/omniscidb-internal/pull/6035) being merged, so the necessary annotation infrastructure is present. This also means that manager-related tests are expected to fail until that is merged.

PS: This also fixes a small unrelated issue with file naming: `rbc/omnisci_backend/column_list.py` was misnamed, not following the pattern of other files, so this renames it to `rbc/omnisci_backend/omnisci_column_list.py`.

### CODE REVIEW

File: rbc/omnisci_backend/omnisci_table_function_manager.py
Comment (guilhermeleobas) [inline rbc/omnisci_backend/omnisci_table_function_manager.py:60]: nit: you can declare the types in the global scope (`i8p = ir.IntType(8).as_pointer()`) and just use them here.
```suggestion
        fnty = ir.FunctionType(i32, [i8p, i8p])
```

File: rbc/omnisci_backend/omnisci_table_function_manager.py
Comment (guilhermeleobas) [inline rbc/omnisci_backend/omnisci_table_function_manager.py:48]: Declare `i8p = ir.IntType(8).as_pointer()` before and just use `i8p`.

File: rbc/omnisci_backend/omnisci_table_function_manager.py
Comment (guilhermeleobas) [inline rbc/omnisci_backend/omnisci_table_function_manager.py:56]: Declare `i8p = ir.IntType(8).as_pointer()` before and just use `i8p`.

File: rbc/omnisci_backend/omnisci_table_function_manager.py
Comment (guilhermeleobas) [inline rbc/omnisci_backend/omnisci_table_function_manager.py:89]: Declare the types before and use `i8p` and `i64` instead. Notice, you can declare all those types in the global scope.

File: rbc/tests/test_omnisci_udtf.py
Comment (guilhermeleobas) [inline rbc/tests/test_omnisci_udtf.py:243]:
```suggestion
        omnisci.sql_execute(
            f'select out0 from table(my_manager_error('
            f'cursor(select f8 from {omnisci.table_name})));')
```

File: rbc/tests/test_omnisci_udtf.py
Comment (guilhermeleobas) [inline rbc/tests/test_omnisci_udtf.py:250]:
```suggestion
    _, result = omnisci.sql_execute(
        f'select out0 from table(my_manager_row_size('
        f'cursor(select f8 from {omnisci.table_name})));')
```

**Review by guilhermeleobas — CHANGES_REQUESTED** [review]
Hi Breno, good work. I have a few suggestions to the PR. Most of them are nits.

File: rbc/omniscidb.py
Comment (pearu) [inline rbc/omniscidb.py:936]: Instead of uses_manager, use function annotations. That is, omniscidb treats the last annotation in annotations list as function annotation. So, insert somewhere above the following code:
```python
function_annotation = dict()  # TODO: retrieve function annotations from orig_sig
if uses_manager:
    function_annotations['uses_manager'] = True
annotations.append(function_annotation)
```

File: rbc/tests/test_omnisci_udtf.py
Comment (pearu) [inline rbc/tests/test_omnisci_udtf.py:227]: Eliminate using `RowMultiplier`.

**Review by pearu — CHANGES_REQUESTED** [review]
Thanks @brenocfg ! The PR looks good. I have some nits and one major request: don't change thrift interface but use function annotation to pass the `use_manager` info to the server.

File: rbc/omniscidb.py
Comment (brenocfg) [reply rbc/omniscidb.py]: Just pushed changes to implement this. See the omnisci PR as well: [PR 6035](https://github.com/omnisci/omniscidb-internal/pull/6035). I tried to implement it in a way that keeps the interfaces for rbc and compile-time UDTFs as close as possible.

File: rbc/tests/test_omnisci_udtf.py
Comment (pearu) [reply rbc/tests/test_omnisci_udtf.py]: CI failures go away when applying this patch.

File: rbc/tests/test_omnisci_udtf.py
Comment (pearu) [inline rbc/tests/test_omnisci_udtf.py:225]:
```suggestion
    omnisci.require_version((5, 9), 'Requires omniscidb-internal PR 6035', label='master')
```

**Review by pearu — CHANGES_REQUESTED** [review]
LGTM! Please apply the omniscidb version suggestions before landing this PR: using the implemented feature from rbc will require omniscidb 5.9 (after the corresponding PR is merged, of course).

File: rbc/omnisci_backend/omnisci_table_function_manager.py
Comment (brenocfg) [reply rbc/omnisci_backend/omnisci_table_function_manager.py:46]: Ahhh, duh! Of course it'd be 5.9, thanks @pearu!

**Review by pearu — APPROVED** [review]
LGTM! Thanks @brenocfg !

Let's merge this after https://github.com/omnisci/omniscidb-internal/pull/6035 lands.

**Review by guilhermeleobas — APPROVED** [review]

**Pull request merged**
